src/ocean/io/model/ISuspendableThrottler.d

```diff
@@ -77,10 +77,11 @@ abstract public class ISuspendableThrottler
         if ( !this.suspendables.contains(s) )
         {
             this.suspendables ~= s;
-            if (this.suspended_)
-            {
-                s.suspend();
-            }
+        }
+
+        if (this.suspended_ && !s.suspended())
+        {
+            s.suspend();
```

Comment:
should it be suspended if it isn't added to the `suspendables`? Anyway if there is any reason to do so then the documentation of the method should be updated.

Reply:
Yes that's the whole point. Recycled instances are in the list and they should be suspended.

Reply:
And good point. I updated the doc accordingly ;).